### DISCUSSION TITLE
I want to create a shortcut to close all windows in the workspace.

### DISCUSSION THREAD

**Discussion by jiggol** [labwc/labwc#2330] [Q&A] [answered]
I would like to set up a shortcut to close all windows in the current workspace or all workspaces at once. Is there a way to do this?

Thank you.

**Comment by Consolatis** [comment, marked as answer]
Using the [ForEach](https://labwc.github.io/labwc-actions.5.html#entry_action_name=foreach) action this should be possible:
```xml
<action name="ForEach">
	<then>
		<action name="Close" />
	</then>
</action>
```

Or, if you are running the master branch, you can also [query the workspace](https://labwc.github.io/labwc-actions.5.html#entry_desktop):
```xml
<action name="ForEach">
	<query desktop="current" />
	<then>
		<action name="Close" />
	</then>
</action>
```

So this should close all windows on all workspaces when pressing `W-c`:
```xml
<keybind key="W-c">
	<action name="ForEach">
		<then>
			<action name="Close" />
		</then>
	</action>
</keybind>
```

**Reply by jiggol** [reply]
Wow, I didn't realize there was a foreach statement. I must have missed it in the manual. Thank you so much!
